### DISCUSSION TITLE
Add pictures of the characters

### DISCUSSION THREAD

**Discussion by Clyptex** [AnimechanOrg/animechan#66] [General]
and if there's any way to access all the characters names   :+1: 
Thanks for the API

**Comment by rocktimsaikia** [comment]
Yes. We can do that and I even thought about the execution before but that would require a lot of time to execute plus it would need a dedicated database that supports media files like [`Cloudinary`](https://cloudinary.com/) which has different pricing plans and since this is just an open-source project I don't want to spend a few extra bucks on adding that feature. Unless someone is willing to back it financially.

**Comment by AndresMorelos** [comment]
@RocktimSaikia What if you send the image as base64 in the response?; you can store it as text in the mongodb document

**Comment by rocktimsaikia** [comment]
@AndresMorelos  That would still require a lot of time. To get the images first then linking to their corresponding characters.  So I don't plan on investing time in adding this feature for now. Although I never tried it with MongoDB before I am sure it would not optimize the images. I only prefer MongoDB for text-based data storage, not for media storage.  Maybe in the future, If I ever get time I will think about it.

**Comment by justanotherbyte** [comment]
@RocktimSaikia why don't you add the images as some sort of link. Like imgur links? If users every needed to extract image data, they could do so within their own code.

**Comment by boneskewer69** [comment]
For a personal project I am using the AniList API. I'm taking the character name I get of a quote by animechan and form a simple GraphQL query.

This is what my query looks like:

```
{
  Character(search: "Levi") {
    name {
      first
      last
      full
      native
    }
    image {
      large
    }
    #description
}
```

Inside the response I get a link to an image of the character. In this case: `https://s4.anilist.co/file/anilistcdn/character/large/b45627-CR68RyZmddGG.png`

Maybe this can help you @ThanksILoveYouWithAllTheRespect.

**Comment by djfjfofoc3** [comment]
just thought copyright law might be worth mentioning, as you'd need permission from each studio/publisher to use the image(s) in question, to be legally allowed to serve it to the end user(s)

**Comment by Nekidev** [comment]
> just thought copyright law might be worth mentioning, as you'd need permission from each studio/publisher to use the image(s) in question, to be legally allowed to serve it to the end user(s)

I don't think studios would bother about having an API that shows their info. Do you really think that MAL/AL/Kitsu/etc ask every studio if they can use the cover?

**Comment by melosh101** [comment]
I have an idea of how this could be done, but I don't want to fetch 1000 images on my own for every character and anime that is in the db, besides it would take some time and effort for me to setup and test,

**Reply by melosh101** [reply]
My idea is to use a locally or externally hosted minio object store. Minio is a s3 compatible object store, that could store the images and cloud be accessible through either the backend or frontend, the object key/link could be stored alongside the quote in our database. So when someone queries for a quote we send a link along side it to the image that then gets served to you through that, but due to the increase of bandwidth going out and the time it takes to find the image I think it would be wise to run a separate ratelimit for the images

**Comment by guruprashanth2004** [comment]
Can find an api which provides these...
